corehq/apps/custom_data_fields/edit_entity.py

```diff
@@ -87,10 +88,10 @@ def get_data_to_save(self):
         return dict(cleaned_data, **system_data)
 
     def init_form(self, post_dict=None):
-        fields = {
-            field.slug: _make_field(field)
-            for field in self.model.get_fields(required_only=self.required_only)
-        }
+        fields = OrderedDict()
```

Comment:
I realize you didn't introduce this issue, but very strange and possibly dangerous that this variable name is reused below.